question_answering_based_on_context_using_openai_and_pinecone/main.bal

```diff
@@ -0,0 +1,98 @@
+import ballerina/http;
+import ballerina/io;
+import ballerina/regex;
+import ballerinax/googleapis.sheets;
+import ballerinax/openai.embeddings;
+import ballerinax/openai.text;
+import ballerinax/pinecone.vector as pinecone;
+
+configurable string sheetsAccessToken = ?;
+configurable string sheetId = ?;
+configurable string sheetName = ?;
+configurable string openAIToken = ?;
+configurable string pineconeKey = ?;
+configurable string pineconeServiceUrl = ?;
+
+final sheets:Client gSheets = check new ({auth: {token: sheetsAccessToken}});
+final text:Client openAIText = check new ({auth: {token: openAIToken}});
+final embeddings:Client openaiEmbeddings = check new ({auth: {token: openAIToken}});
+final pinecone:Client pineconeClient = check new ({apiKey: pineconeKey}, serviceUrl = pineconeServiceUrl);
+
+const MAXIMUM_NO_OF_DOCS = 10;
+const NAMESPACE = "ChoreoDocs";
+
+service / on new http:Listener(8080) {
+    function init() returns error? {
+        sheets:Range range = check gSheets->getRange(sheetId, sheetName, "A2:B");
+        pinecone:Vector[] vectorArray = [];
+
+        foreach any[] row in range.values {
+            string title = <string>row[0];
+            string content = <string>row[1];
+            float[] vector = check getEmbedding(string `${title} ${"\n"} ${content}`);
+            vectorArray[vectorArray.length()] = {id: title, values: vector, metadata: {"content": content}};
+        }
+
+        pinecone:UpsertRequest req = {vectors: vectorArray, namespace: NAMESPACE};
+        pinecone:UpsertResponse response = check pineconeClient->/vectors/upsert.post(req);
```

Comment:
```suggestion
        pinecone:UpsertResponse response = check pineconeClient->/vectors/upsert.post({vectors, namespace: NAMESPACE});
```

Reply:
Addressed in https://github.com/ballerina-guides/ai-samples/pull/66/commits/f4551b3b1b789c9eca43052f52d16841f7f89fbd.